crmsh/utils.py

```diff
@@ -3028,15 +3022,30 @@ def check_no_quorum_policy_with_dlm():
         logger.warning("The DLM cluster best practice suggests to set the cluster property \"no-quorum-policy=freeze\"")
 
 
-def set_property_conditionally(property_name, value_from_calculation):
+def set_property(property_name, property_value, property_type="crm_config", property_id=None, conditional=False):
     """
-    Set cluster property if calculated value is larger then current cib value
+    Set property for cluster, resource and operator
+
+    "property_type" can be crm_config|rsc_defaults|op_defaults
+    "property_id" can be existing meta_attributes id
+    When "conditional" is True, set the property if given "property_value" is larger then value from cib
     """
-    _value = get_property(property_name)
-    value_from_cib = int(_value.strip('s')) if _value else 0
-    if value_from_cib < value_from_calculation:
-        cmd = "crm configure property {}={}".format(property_name, value_from_calculation)
-        get_stdout_or_raise_error(cmd)
+    origin_value = get_property(property_name, property_type)
+    if origin_value and str(origin_value) == str(property_value):
+        return
+    if conditional:
+        if crm_msec(origin_value) >= crm_msec(property_value):
+            return
+    if property_id:
+        out = get_stdout_or_raise_error("crm configure show")
+        if not re.search(" {}:".format(property_id), out):
+            logger.warning("No such id \"{}\" exist".format(property_id))
+            property_id = None
```

Comment:
AFAIU, this way, it'll basically let crm_attribute create the set with its convention. But OTOH, any invocations of `crm configure rsc_defaults` won't use this set anyways, right?  Not sure if we could make everything sensible with this PR,  otherwise we could still stick with your previous way. Then an improvement in the future could be to make both "crm configure" and "crm bootstrap" consistently follow crm_attribute's convention.

Reply:
I think here calling `crm configure rsc_defaults` is better, since in cibconfig.py, already did a lots of work, if the `rsc-option` not exist, `rsc-option` will be created and make sure all rsc_defaults' properties are under this id

Reply:
Already changed to use `crm configure property|rsc_defaults`